### PR TITLE
Add JITRuntime test

### PR DESCRIPTION
**Pull request by asi1024**
~Depends on #400 and #405~

TODO: need to be fixed after supporting intermediate variable comparison in Comparer. (added by @kmaehashi)

### CODE REVIEW

**Comment by asi1024** [comment]
Rebased. PTAL.
/test

**Review by kmaehashi — CHANGES_REQUESTED** [review]
LGTM except nits!

File: tests/pytorch_pfn_extras_tests/runtime_tests/test_jit_runtime.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                    **{str(i): x for i, x in enumerate(out[:-inter_size])},
+                    **{name: x for name, x in zip(self._names, out[-inter_size:])},
+                )
+
+            new_forward = self.forward
+            self.forward = self._orig_forward
+
+            with trace(self) as (new_module, outputs):
+                warnings.simplefilter('ignore')
+                self._traced_mod = torch.jit.trace_module(new_module, {"forward": args})
```
Comment (asi1024) [inline tests/pytorch_pfn_extras_tests/runtime_tests/test_jit_runtime.py:33]: Fixes https://github.com/pfnet/pytorch-pfn-extras/pull/406#discussion_r748066213.
```suggestion
                warnings.simplefilter('ignore')
                self._traced_mod = torch.jit.trace_module(new_module, {"forward": args})
                warnings.simplefilter('error')
```

**Review by kmaehashi — DISMISSED** [review]

File: tests/pytorch_pfn_extras_tests/runtime_tests/test_jit_runtime.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                    **{name: x for name, x in zip(self._names, out[-inter_size:])},
+                )
+
+            new_forward = self.forward
+            self.forward = self._orig_forward
+
+            with trace(self) as (new_module, outputs):
+                with warnings.catch_warnings():
+                    warnings.simplefilter("ignore")
+                    self._traced_mod = torch.jit.trace_module(
```
Comment (kmaehashi) [inline tests/pytorch_pfn_extras_tests/runtime_tests/test_jit_runtime.py:34]: For record:
* Due to PyTorch bug, this raises `SystemError` when `filterwarnings = error` is set in `pytest.ini`.
* Tried to add `ignore::torch.jit.TracerWarning:torch.(?!jit)` to the filter to emulate https://github.com/pytorch/pytorch/blob/v1.10.0/torch/jit/_trace.py#L531-L540, however this broke EnsureShape JIT test for some reason. (#430)
* `catch_warnings` is thread-unsafe but we accept it for now as we don't have a plan to use pytest-xdist.

**Review by kmaehashi — APPROVED** [review]

**Pull request merged**